src/sagemaker/estimator.py

```diff
@@ -675,8 +677,18 @@ def __init__(
             sagemaker_session=self.sagemaker_session,
         )
 
-        self.profiler_config = profiler_config
-        self.disable_profiler = disable_profiler
+        self.profiler_config = resolve_value_from_config(
+            direct_input=profiler_config,
+            config_path=TRAINING_JOB_PROFILE_CONFIG_PATH,
+            default_value=None,
+            sagemaker_session=self.sagemaker_session,
+        )
```

Comment:
Seems profiler_config is a class sagemaker.debugger.ProfilerConfig ?

So pulling the dict from the config and setting it directly might lead to unexpected behavior. Do we need this injection anyway? Later as more params are added to the profiler_config, I think the existing class helper could be used

Reply:
Makes sense. Great callout - I can kick it out and only inject the disable_profiler value in.

Reply:
`resolve_class_attribute_from_config` is the right choice here.